conformance/lang/expressions/list-constructor-expr/spread_operator.balt

```diff
@@ -0,0 +1,728 @@
+    io:println(v6); // @output [1,2]
+    io:println(v7); // @output [1,2]
+    io:println(v8); // @output [1,2,3]
+    io:println(v9); // @output ["x",1]
+    io:println(v10); // @output [1,2]
+}
+
+Test-Case: output
+Description: Test tuple type with spread operator.
+Labels: list-constructor-expr, spread-operator, tuple-type, var, never-type, intersection-type,
+        boolean-type, var
+
+type Foo int[2] & readonly;
+type Bar int[2]|string|(readonly & int[3])|float;
+
+[int, string, never...] & readonly a1 = [1, "x"];
+[int, int, never...] & readonly a2 = [1, 2];
+[never...] & readonly a3 = [];
+[int, boolean] & readonly a4 = [1, true];
+int[2] & readonly a5 = [1, 2];
+Foo a6 = [1, 2];
+Bar a7 = [1, 2, 3];
+[readonly, readonly] a8 = ["x", 1];
+var a10 = [1, 2];
+
+[int, string] v0 = [...a1];
+[int, int] v1 = [...a2];
+[never...] v2 = [...a3];
+[int, string, int, int, int, boolean] v3 = [...a1, ...a2, ...a4];
+[int, string, int, int, int, boolean] v4 = [...a1, ...a2, ...a3, ...a4];
+var v5 = [1, 2, ...a1, ...a2, ...a3, ...a4, ...[1, 1]];
+int[] v6 = [...a5];
+int[] v7 = [...a6];
+int[3] v8 = [...<readonly & int[3]>a7];
+readonly[2] v9 = [...a8];
+readonly[] v10 = [...a10];
+
+function init() {
+    io:println(v0); // @output [1,"x"]
+    io:println(v1); // @output [,1,2]
+    io:println(v2); // @output []
+    io:println(v3); // @output [1,"x",1,2,true]
+    io:println(v4); // @output [1,"x",1,2,true]
+    io:println(v5); // @output [1,2,1,"x",1,2,1,true,1,1]
+    io:println(v6); // @output [1,2]
+    io:println(v7); // @output [1,2]
+    io:println(v8); // @output [1,2,3]
+    io:println(v9); // @output ["x",1]
+    io:println(v10); // @output [1,2]
+}
+
+Test-Case: output
+Description: Test spread operator with any or anydata contextually expected type.
+Labels: list-constructor-expr, spread-operator, tuple-type, var, never-type, intersection-type,
+        boolean-type, any-type, anydata-type
+
+type Foo any[2];
+type Bar [int, int, int];
+
+function init() {
+    [int, int] b1 = [1, 1];
+    int[2] b2 = [2, 2];
+    [int, int, int] b3 = [3, 3, 3];
+    [int, boolean] b4 = [1, true];
+    any a1 = b1;
+    anydata a2 = b2;
+    [any, any] a4 = b4;
+    any[2] a5 = b2;
+    Foo a6 = b2;
+    [int, int, int] a7 = b3;
+    [anydata, anydata] a8 = ["x", 1];
+    [int, int] v0 = [...<[int, int]>a1];
+    [int...] v1 = [...<int[2]>a2];
+    anydata v3 = [...<[int, int]>a1, ...<int[2]>a2, ...<[int, boolean]>a4];
+    any[] v6 = [...a5];
+    any[2] v7 = [...a6];
+    anydata v8 = [...<int[3]>a7];
+    anydata v9 = [...a8];
+    io:println(v0); // @output [1,1]
+    io:println(v1); // @output [2,2]
+    io:println(v3); // @output [1,1,2,2,1,true]
+    io:println(v6); // @output [2,2]
+    io:println(v7); // @output [2,2]
+    io:println(v8); // @output [3,3,3]
+    io:println(v9); // @output ["x",1]
+}
+
+Test-Case: output
+Description: Test spread operator without contextually expected type.
```

Comment:
This description is repeated 6 times. Shall we merge similar tests? 
if there are different shall we specify them in the description?